src/sqlfluff/dialects/dialect_tsql.py

```diff
@@ -1686,13 +1685,44 @@ class SetStatementSegment(BaseSegment):
                         ),
                     ),
                 ),
+                Sequence(
+                    Ref("ParameterNameSegment"),
+                    Ref("AssignmentOperatorSegment"),
+                    Ref("ExpressionSegment"),
+                ),
             ),
         ),
         Dedent,
         Ref("DelimiterGrammar", optional=True),
     )
 
 
+class AssignmentOperatorSegment(BaseSegment):
+    """One of the assignment operators.
+
+    Includes simpler equals but also +=, -=, etc.
+    """
+
+    type = "assignment_operator"
+    match_grammar = OneOf(
+        Ref("EqualsSegment"),
+        Sequence(
+            OneOf(
+                Ref("PlusSegment"),
+                Ref("MinusSegment"),
+                Ref("DivideSegment"),
+                Ref("MultiplySegment"),
+                Ref("ModuloSegment"),
+                Ref("BitwiseAndSegment"),
+                Ref("BitwiseOrSegment"),
+                Ref("BitwiseXorSegment"),
+            ),
+            Ref("EqualsSegment"),
+            allow_gaps=False,
```

Comment:
Is this line necessary? While I would prefer it, we did have a previous PR (see #1767) to allow gaps in comparison operators because apparently TSQL supports that. Does it support spaces here too?

Reply:
Judging from documentation, it does not seem to support spaces in compound assignment operators.

https://docs.microsoft.com/en-us/sql/t-sql/language-elements/set-local-variable-transact-sql?view=sql-server-ver15

But the same seems to be true with the documentation of comparison operators. Didn't know it supported that.

Anyway, on a real server
`SELECT 1 WHERE 1 > = 0`
works, but
`SET @t + = 1`
does not. So we need `allow_gaps=False` here.

Reply:
Cool, cheers for checking!